homeassistant/components/guardian/binary_sensor.py

```diff
@@ -23,56 +28,70 @@
     DOMAIN,
     SIGNAL_PAIRED_SENSOR_COORDINATOR_ADDED,
 )
+from .util import GuardianDataUpdateCoordinator
 
 ATTR_CONNECTED_CLIENTS = "connected_clients"
 
 SENSOR_KIND_AP_INFO = "ap_enabled"
 SENSOR_KIND_LEAK_DETECTED = "leak_detected"
 SENSOR_KIND_MOVED = "moved"
 
-SENSOR_DESCRIPTION_AP_ENABLED = BinarySensorEntityDescription(
-    key=SENSOR_KIND_AP_INFO,
-    name="Onboard AP enabled",
-    device_class=BinarySensorDeviceClass.CONNECTIVITY,
-    entity_category=EntityCategory.DIAGNOSTIC,
-)
-SENSOR_DESCRIPTION_LEAK_DETECTED = BinarySensorEntityDescription(
-    key=SENSOR_KIND_LEAK_DETECTED,
-    name="Leak detected",
-    device_class=BinarySensorDeviceClass.MOISTURE,
-)
-SENSOR_DESCRIPTION_MOVED = BinarySensorEntityDescription(
-    key=SENSOR_KIND_MOVED,
-    name="Recently moved",
-    device_class=BinarySensorDeviceClass.MOVING,
-    entity_category=EntityCategory.DIAGNOSTIC,
-)
+
+@dataclass
+class ValveControllerBinarySensorDescription(
+    BinarySensorEntityDescription, ValveControllerEntityDescription
+):
+    """Describe a Guardian valve controller binary sensor."""
+
 
 PAIRED_SENSOR_DESCRIPTIONS = (
-    SENSOR_DESCRIPTION_LEAK_DETECTED,
-    SENSOR_DESCRIPTION_MOVED,
+    BinarySensorEntityDescription(
+        key=SENSOR_KIND_LEAK_DETECTED,
+        name="Leak detected",
+        device_class=BinarySensorDeviceClass.MOISTURE,
+    ),
+    BinarySensorEntityDescription(
+        key=SENSOR_KIND_MOVED,
+        name="Recently moved",
+        device_class=BinarySensorDeviceClass.MOVING,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
 )
+
 VALVE_CONTROLLER_DESCRIPTIONS = (
-    SENSOR_DESCRIPTION_AP_ENABLED,
-    SENSOR_DESCRIPTION_LEAK_DETECTED,
+    ValveControllerBinarySensorDescription(
+        key=SENSOR_KIND_LEAK_DETECTED,
+        name="Leak detected",
+        device_class=BinarySensorDeviceClass.MOISTURE,
+        api_category=API_SYSTEM_ONBOARD_SENSOR_STATUS,
+    ),
+    ValveControllerBinarySensorDescription(
+        key=SENSOR_KIND_AP_INFO,
+        name="Onboard AP enabled",
+        device_class=BinarySensorDeviceClass.CONNECTIVITY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        api_category=API_WIFI_STATUS,
+    ),
 )
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up Guardian switches based on a config entry."""
+    paired_sensor_coordinators = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR_PAIRED_SENSOR
+    ]
+    valve_controller_coordinators = hass.data[DOMAIN][entry.entry_id][DATA_COORDINATOR]
```

Comment:
```suggestion
    entry_data = hass.data[DOMAIN][entry.entry_id]
    paired_sensor_coordinators = entry_data[DATA_COORDINATOR_PAIRED_SENSOR]
    valve_controller_coordinators = entry_data[DATA_COORDINATOR]
```

I'd probably make `entry_data` a dataclass in a followup